core/services/subscription.go

```diff
@@ -1,271 +1,170 @@
-	strpkg "github.com/smartcontractkit/chainlink/core/store"
 	"github.com/smartcontractkit/chainlink/core/store/models"
-	"github.com/smartcontractkit/chainlink/core/store/presenters"
-	"github.com/smartcontractkit/chainlink/core/utils"
-
-	ethereum "github.com/ethereum/go-ethereum"
-	"github.com/pkg/errors"
-	"go.uber.org/multierr"
 )
 
-// Unsubscriber is the interface for all subscriptions, allowing one to unsubscribe.
-type Unsubscriber interface {
-	Unsubscribe()
-}
-
-// JobSubscription listens to event logs being pushed from the Ethereum Node to a job.
+// JobSubscription is the interface for listening for requests from the Ethereum node for a particular job.
+// The only thing you can do with it is Unsubscribe.
 type JobSubscription struct {
-	Job           models.JobSpec
-	unsubscribers []Unsubscriber
-}
-
-// StartJobSubscription constructs a JobSubscription which listens for and
-// tracks event logs corresponding to the specified job. Ignores any errors if
-// there is at least one successful subscription to an initiator log.
-func StartJobSubscription(job models.JobSpec, head *models.Head, store *strpkg.Store, runManager RunManager) (JobSubscription, error) {
-	var merr error
-	var unsubscribers []Unsubscriber
-
-	initrs := job.InitiatorsFor(models.LogBasedChainlinkJobInitiators...)
-
-	nextHead := head.NextInt() // Exclude current block from subscription
-	if replayFromBlock := store.Config.ReplayFromBlock(); replayFromBlock >= 0 {
-		replayFromBlockBN := big.NewInt(replayFromBlock)
-		nextHead = replayFromBlockBN
-	}
-
-	for _, initr := range initrs {
-		unsubscriber, err := NewInitiatorSubscription(initr, store.EthClient, runManager, nextHead, ReceiveLogRequest)
-		if err == nil {
-			unsubscribers = append(unsubscribers, unsubscriber)
+	// Job needs to be public for callers of StartJobSubscription
+	Job            models.JobSpec
+	logBroadcaster logBroadcasterRegistration
+	runManager     runManagerCreate
+	listeners      []*listener
+	logger         jsLogger
+}
+
+func StartJobSubscription(job models.JobSpec, logBroadcaster logBroadcasterRegistration, runManager runManagerCreate, logger jsLogger) (JobSubscription, error) {
+	js := JobSubscription{
+		Job:            job,
+		logBroadcaster: logBroadcaster,
+		runManager:     runManager,
+		logger:         logger,
+	}
+	js.listeners = js.makeAndRegisterListeners()
+	if len(js.listeners) == 0 {
+		return js, fmt.Errorf("unable to subscribe to any logs, check earlier errors in this message, and the initiator types, jobID=%s", job.ID)
+	}
+	return js, nil
+}
+
+func (js JobSubscription) makeAndRegisterListeners() []*listener {
+	initiators := js.Job.InitiatorsFor(models.LogBasedChainlinkJobInitiators...)
+	listeners := []*listener{}
+	for _, initiator := range initiators {
+		l := &listener{
+			jobID:      js.Job.ID,
+			initiator:  &initiator,
+			runManager: js.runManager,
+			logger:     js.logger,
+		}
+		if !js.logBroadcaster.Register(initiator.InitiatorParams.Address, l) {
```

Comment:
Warning! This will hang forever if the log broadcaster is not started. Are we ok with that or ought we to panic? Return error?

Reply:
That's a good point. Tbh, I don't know. 
Afaict, all the "head trackables" (which includes the JobSubscriber) are `Start()`ed after the LogBroadcaster.
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/chainlink/application.go#L174-L182